### DISCUSSION TITLE
Colouring Cities Developer Calls

### DISCUSSION THREAD

**Discussion by tomalrussell** [tomalrussell/colouring-london#799] [Q&A]
Just to start the discussion about sync calls, open to the wider Colouring Cities developer group, and potentially other interested people who've found their way here.

Suggest:
- bi-monthly (every 8 weeks from 18 May 2022, 7-8am GMT)
- do agenda-building as the first action of the call
- keep notes in an etherpad or similar, tidy and publish these after each call

For example, topics for early calls could be deployment of Colouring Cities platforms and setting up development environments, shared development needs including translation, location-specific configuration of the application.

cc @colouring-cities/everyone

**Comment by matkoniecz** [comment]
Posting here to indicate that I would be happy to join.

**Comment by edwardchalstrey1** [comment]
Several thoughts on the above based on internal discussion:
 
1. An initial session at 7am GMT/UTC (so 8am BST?) to involve people as far afield as Australia (UTC+11?) and everywhere else on a single call to touch base and discuss our plans makes sense, however I would encourage subsequent sessions to be during the EMEA working day or to be on an ad hoc basis with those further afield
2. I think the first session we (Colouring London team) could encourage attendees to look at [the dev environment setup](https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-dev-environment.md) in their own time post-meeting (if any of you read this, please do check out in advance), and just spend some time explaining what this is during the meeting, but otherwise focus on introductions.
3. Perhaps a subsequent session (e.g. ~2 weeks after rather than 8 and in a 2hr slot) I could run a workshop for anyone who wanted help with setting up a Colouring London dev environment. It would be great to have the outcome of this session to be someone from each Colouring Cities project having a working Colouring London setup on a VM - I'm sure this will be extremely valuable when developing their own version for their own city/country.
4. We can assign someone to take some minutes from those meetings and put them somewhere (e.g. here on GitHub) - I can do this, or we can use a collaborative tool such as HackMD (or Etherpad as you suggest!)
5. I'd encourage asynchronous discussion to be the default and focus on this GH discussion board becoming the main channel of developer collaboration (but also to make people aware of https://discuss.colouring.london/ for more general discussion)

**Comment by tomalrussell** [comment]
Brief notes from previous call:

## Dev, ci/cd deployment
- Ubuntu 20.04 - errors through npm install - suggest "npm ci" to follow lockfile
- Ali update to 20.04 with containerisation - link to dockerised branch?
- Ed update to dev environment setup docs - [https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-dev-environment.md](https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-dev-environment.md)

## Data loading
- how to convert and load to the database
- ETL specific to OS data [https://github.com/colouring-cities/colouring-london/tree/master/etl](https://github.com/colouring-cities/colouring-london/tree/master/etl)
- first:  run the migrations to set up database schema
- PostGIS extension (for spatial data) see [http://postgis.net/workshops/postgis-intro/geometries.html](http://postgis.net/workshops/postgis-intro/geometries.html)
- Geometry Type is 'Polygon', Well-known-Text

## Data merging
- government source / OSM merging for Indonesia
- MS building footprints (good for rural areas) / OSM (good for cities) for Australia. Try to link with GNAF(?) address record for Australia
- Athens - future matching of government/OSM data. Per municipality, visual inspection and choose preferred source.
- update process to geometries
- data attributes from OSM - land use? in Athens
- data source iconography

## Multilingual support
- database text elements (enums to reference tables), use ID then pick text in frontend
- React library for translation, keep text strings in JSON format, switch using UI toggle
- some API addition to fetch translations
- use ID for styling instead of text value
- some object values coded in a hard-to-maintain way

**Comment by tomalrussell** [comment]
Brief notes from recent call:

## Workshop

Theodor and Colouring Dresden team are taking part in a funding competition for citizen science programmes - workshop today talking with stakeholders to collect ideas about what might be the most interesting topics/aspects (energy, building materials, ...) of Colouring Dresden platform/project.



## Mapathon user feedback

Zara and Colouring Indonesia team recently ran a mapathon over ~4days, ~30 participants, tested beta version, crowdsourcing, shared or discussed data that might be suitable for bulk upload.



Asked: What should be done differently? How easy is it the site to use, is it accessible?



### Difficulty of identifying buildings

   * start centred on a pin on the national monument, always land on same map view
   * search function would be really useful - avoid scroll and scroll to get to area of interest
       * consider built-in search API/table (used for postcode search in London)
       * consider Nominatim API


### Background map

   * sometimes discrepancy between building footprints and background map
   * look at google streetview or similar for orientation, switch back to colouring
       * satellite imagery or other basemap would be useful


### Map scale

   * Platform has no scale visible/defined
   * zoom level is used in web mapping but not displayed on the platform
   * need to guess that the map is at a particular scale by eye
       * show scale - 1:10000, zoom level, line showing distance on the map


### Mobile view

   * could be doing crowdsourcing on the move
   * not as convenient as laptop
   * things to make the mobile version easier to navigate
       * scroll up on menu - top categories are hidden under header
       * some elements a bit fiddly - suggestion to all use the mobile view and pay attention to design and interactions to fix the easiest things


### Categories and data fields

   * how easy is it to switch data categories? - some consistency is desirable
   * e.g. BREEAM not used in Indonesia, other specific certification (Green Mark)
   * specifically Type, Sustainability, StreetScape, Planning raise lots of differences
   * how easy is it to switch things
       * thinking through shared/different categories
       * approach to code sharing - could implement in forks, or have long list in "core" platform
       * colouring cities data model? Do we have a long list of attributes, enable/disable as necessary?
       * probably one to have a longer discussion about with wider group


## Documentation setup

Mohammed and Colouring Bahrain have been setting up a beta site, running through production setup.

   * snags: postgres 12 > 14, node hard-coded 12.14.1 throughout, npm no longer use next
   * add note on option to import from database dump - or link to data importing/migrations docs
   * developing shell script to run full setup
       * Mohammed will make a PR

**Comment by polly64** [comment]
At next CCRP PI meeting, working groups for enhancement of specific data categories will be proposed and discussed. This will allow us to address issues of a) core data subcategories for which formatting will be identical across CCRP platform regardless of country- allowing for easy joint working on data analysis at international scale , b) core data subcategories for which consistent formatting more complex, c) additional subcategories unique to specific countries/global regions but for which open code needs to be accessible.  (NB some Colouring London prototype subcategories and features have not been activated simply because of lack of engineering time available. Increased joint working of s/e and PI teams will help accelerate this process).
